src/Storages/MergeTree/MergeTreeIndexText.cpp

```diff
@@ -399,49 +337,62 @@ DictionaryBlock deserializeDictionaryBlock(ReadBuffer & istr)
 
     std::vector<TokenPostingsInfo> token_infos;
     token_infos.reserve(num_tokens);
+    using enum PostingsSerialization::Flags;
 
     for (size_t i = 0; i < num_tokens; ++i)
     {
-        UInt64 header;
-        UInt32 cardinality;
+        TokenPostingsInfo info;
 
-        readVarUInt(header, istr);
-        readVarUInt(cardinality, istr);
+        readVarUInt(info.header, istr);
```

Comment:
Can we make static function `deserialize/serializeTokenPostingsInfo()` which we call from here?